### ISSUE TITLE
Map plot adding Stereographic so i can show the milky way and constellation borders

### ISSUE THREAD

**Issue by bathoorn** [steveberardi/starplot#50]
I would like to see the milky way and the constellation borders in the ZenithPlot. Started adding constellation borders but then got to thinking. Maybe i can add Stereograohic projection to the MapPlot and do everything there.

I hacked Stereographic and Orthographic projections into your implementation. And was able to produce these images

stereographic
<img width="2073" alt="04_map_stereographic" src="https://github.com/steveberardi/starplot/assets/760615/5ec25cf2-da4a-46cc-8452-8523e1e7ac53">

orthographic
<img width="1953" alt="05_map_orthographic" src="https://github.com/steveberardi/starplot/assets/760615/49fcb972-f3c1-46d5-aee4-8639f6e2e374">

I would like to share the code if you are interested. Right now it is a bit of a mess but i could fork this repo and work on a merge request.

Thanks for the great project i already had a lot of fun with it

**Comment by steveberardi** [comment]
@bathoorn 

Thanks for your feedback and sharing those great looking plots! 😃 

I’d love to add support for plotting the Milky Way and constellation borders on ZenithPlots. There’s some shared logic across plot types for [plotting polygons](https://github.com/steveberardi/starplot/blob/main/src/starplot/base.py#L300-L318) which could help with this. Ideally, the same code could be used to plot the Milky Way for all plot types.

I also like the idea of adding support for Orthographic projections to MapPlots. I'd be curious to see your code for this. Hopefully it was as easy as adding Orthographic to the [projection enum](https://github.com/steveberardi/starplot/blob/main/src/starplot/map.py#L32)?

That’d be great if you want to share some of your code through a PR :) I don’t have any formal contributor guide yet, but definitely open to contributions. I'm working on refactoring some of the shared functionality for plots on the `v0.9.0` branch, so if you wanna create a branch from there that'd help us avoid conflicts on the PR.

Thanks again!

Steve

**Comment by bathoorn** [comment]
i created a merge request for adding constellation boundaries to ZenithPlot

You can test it through running example_01.py

It should produce this

<img width="1109" alt="01_star_chart" src="https://github.com/steveberardi/starplot/assets/760615/aac530e4-8460-45ad-9fe6-70a4a4283773">

**Comment by steveberardi** [comment]
thanks! I'll review the PRs sometime this weekend :) One thing I noticed in your example above though is that not all constellation lines are dashed. I'm guessing cause the data file has overlapping borders. I ran into a similar problem when trying to refactor the constellation border plotting, and used shapely's [`unary_union`](https://github.com/steveberardi/starplot/blob/8dc4a8a106041c59447b4d81ceffdae5060bc99d/src/starplot/map.py#L395) to merge the lines

**Comment by bathoorn** [comment]
Hi Steve,

Indeed this is because i interpolate and send all the points to the
projection. And there a lot of short line segments interfere with the
dashes.
But it should also be possible to draw lines in ra dec space and then
project to something else. I think that is what happens in the ZENITH type
i added to the MapPlot projections. Because there it looks ok.

**Comment by bathoorn** [comment]
I also ran example_08 09 and 10

ORTHGRAPHIC
<img width="741" alt="08_orthographic" src="https://github.com/steveberardi/starplot/assets/760615/93d45489-35df-43ae-b5cf-502b06ebe3e7">

STEREOGRAPHIC
<img width="581" alt="09_stereographic" src="https://github.com/steveberardi/starplot/assets/760615/acfd83a8-fc8b-4168-aa30-cd0aa436d7ed">

ZENITH
<img width="761" alt="10_zenith" src="https://github.com/steveberardi/starplot/assets/760615/fd5c4fc5-fd0a-402d-988e-abea2d5aaa5a">

**Comment by steveberardi** [comment]
@bathoorn I'm getting ready to release the next version of starplot, `0.9.0`, and I have your projection contributions in the [open PR for it](https://github.com/steveberardi/starplot/pull/57) -- thanks so much again for helping with this!

In the new version, I also made some changes to how you plot things and what's visible -- mainly, I removed the `visible` style property and now instead you control what's plotted by explicitly calling functions on the plot instance like `p.stars(mag=12)` -- I feel like this gives you a lot more control for plotting different things in different ways, and it's more consistent with other plotting frameworks (i.e. you start with an empty plot). Anyway, I'd greatly appreciate any thoughts or concerns you have about the changes in the v0.9.0 PR - thanks!!
